### DISCUSSION TITLE
Deleted Github branch

### DISCUSSION THREAD

**Discussion by ANGSHUK1** [community/community#54432] [New to GitHub]
HI, I deleted a github repository recently and I restored it. However there was a branch in the github repo that i deleted by mistake and want to restore it back. However I cannot see it. 

Is there any way of getting it back?

**Comment by hrishikesh-kocharekar-dev** [comment]
If you have deleted a branch from a GitHub repository and then restored the repository, the deleted branch may not be visible in the repository anymore. However, it is possible to restore the deleted branch using Git.

First, you need to find the commit hash of the last commit on the deleted branch. You can do this by running the command:
`git reflog`

This will show a list of all the recent changes to the repository, including the commit hash of the last commit on the deleted branch.

Once you have the commit hash, you can recreate the deleted branch by running the command:
`git branch <branch-name> <commit-hash>`

Replace `<branch-name>` with the name of the deleted branch and `<commit-hash>` with the commit hash you obtained from the `git reflog` command.

After running this command, the deleted branch should be restored and you should be able to see it in the repository.

Note that if the branch was deleted a long time ago and there have been many new commits since then, it may be difficult or impossible to restore the branch, as the necessary commit history may have been lost.

**Reply by airtower-luna** [reply]
That would work if the branch had been deleted from a *local* repository recently, not for a branch that existed only on GitHub (where you don't get shell access).

**Reply by SamuelBL101** [reply]
what to do then?

**Comment by DhrubajyotiBiswasfounderofDribll** [comment]
Yes, you should be able to do `git reflog --no-abbrev` and find the` SHA1` for the commit at the tip of your deleted branch, then just` git checkout [sha]`. And once you're at that commit, you can just `git checkout -b [branchname]` to recreate the branch from there.
